applications/FluidDynamicsApplication/tests/embedded_piston_test.py

```diff
@@ -0,0 +1,223 @@
+
+        # Recompute the level-set function
+        x_0 = 2.5
+        for node in self._GetSolver().GetComputingModelPart().Nodes:
+            node.SetSolutionStepValue(KratosMultiphysics.DISTANCE, node.X0 - x_0 - dis)
+
+        # Apply base BCs
+        # Note it is done after the level-set update to correct it with the distance modification process
+        super(CustomFluidDynamicsAnalysis,self).ApplyBoundaryConditions()
+
+        # Set the EMBEDDED_VELOCITY
+        for element in self._GetSolver().GetComputingModelPart().Elements:
+            # Check if the element is split
+            n_pos = 0
+            n_neg = 0
+            for node in element.GetNodes():
+                if node.GetSolutionStepValue(KratosMultiphysics.DISTANCE) < 0.0:
+                    n_neg += 1
+                else:
+                    n_pos += 1
+            # If it is split, set the EMBEDDED_VELOCITY
+            if n_neg != 0 and n_pos != 0:
+                element.SetValue(KratosMultiphysics.EMBEDDED_VELOCITY, [vel,0.0,0.0])
+            else:
+                element.SetValue(KratosMultiphysics.EMBEDDED_VELOCITY, [0.0,0.0,0.0])
+
+    def OutputSolutionStep(self):
+        if self.print_output:
+            super(CustomFluidDynamicsAnalysis,self).OutputSolutionStep()
+
+
+@UnitTest.skipUnless(have_mesh_moving,"Missing required application: MeshMovingApplication")
+class EmbeddedPistonTest(UnitTest.TestCase):
+
+    # Embedded element tests
+    def testEmbeddedPiston2D(self):
+        self.A = 1.5
+        self.w = 2.0 * math.pi
+        self.work_folder = "EmbeddedPiston2DTest"
+        self.reference_file = "reference_embedded_piston_2D"
+        self.settings = "EmbeddedPiston2DTestParameters.json"
+        self.ExecuteEmbeddedPistonTest()
+
+    def ExecuteEmbeddedPistonTest(self):
+        with UnitTest.WorkFolderScope(self.work_folder, __file__):
+            self.setUp()
+            self.setUpProblem()
+            self.runTest()
+            self.tearDown()
+            self.checkResults()
+
+    def setUp(self):
+        self.check_tolerance = 1e-6
+        self.print_output = False
+        self.print_reference_values = False
+
+    def tearDown(self):
+        with UnitTest.WorkFolderScope(self.work_folder, __file__):
+            try:
+                filename_fluid = self.ProjectParameters["solver_settings"]["model_import_settings"]["input_filename"].GetString()
+                filename_structure = filename_fluid + '_structure'
+                KratosUtilities.DeleteFileIfExisting(filename_fluid + '.time')
+                KratosUtilities.DeleteFileIfExisting(filename_structure + '.time')
+                if not self.print_output:
+                    KratosUtilities.DeleteFileIfExisting(filename_fluid + '.post.bin')
+                    KratosUtilities.DeleteFileIfExisting(self.work_folder + '.post.lst')
+            except FileNotFoundError as e:
```

Comment:
I don't think you need the try-except here, DeleteFileIfExisting already has this builtin

Reply:
Removed! Thanks for the tip.